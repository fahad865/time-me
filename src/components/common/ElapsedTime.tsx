import * as React from 'react';

export interface Props {
  elapsedTime: number;
}

function ElapsedTime({ elapsedTime }: Props) {
  const { hours, minutes, seconds } = getTimeUnits(elapsedTime);
  return (
    <div style={{ margin: '5px' }}>
      <span>{padLeft(hours)}:</span>
      <span>{padLeft(minutes)}:</span>
      <span>{padLeft(seconds)}</span>
    </div>
  );
}

function padLeft(value: string) {
  return ('0' + value).slice(-2);
}

function getTimeUnits(timeInSeconds: number) {
  return {
    hours: Math.floor(timeInSeconds / 3600).toString(),
    minutes: Math.floor((timeInSeconds / 60) % 60).toString(),
    seconds: Math.floor((timeInSeconds % 60) % 60).toString()
  };
}
export default ElapsedTime;