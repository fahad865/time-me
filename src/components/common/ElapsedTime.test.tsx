import * as React from 'react';
import { shallow } from 'enzyme';
import ElapsedTime from './ElapsedTime';

describe('Elapsed Time', () => {
  it('it should render correctly', () => {
    expect(shallow(<ElapsedTime elapsedTime={2000} />)).toMatchSnapshot();
  });
});