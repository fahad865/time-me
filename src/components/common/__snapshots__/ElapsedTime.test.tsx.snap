// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elapsed Time it should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ElapsedTime
    elapsedTime={2000}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          00
          :
</span>,
        <span>
          33
          :
</span>,
        <span>
          20
</span>,
      ],
      "style": Object {
        "margin": "5px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "00",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "00",
          ":",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "33",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "33",
          ":",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "20",
        },
        "ref": null,
        "rendered": "20",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            00
            :
</span>,
          <span>
            33
            :
</span>,
          <span>
            20
</span>,
        ],
        "style": Object {
          "margin": "5px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "00",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "00",
            ":",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "33",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "33",
            ":",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "20",
          },
          "ref": null,
          "rendered": "20",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
